include/systems/variational_smoother_constraint.h

```diff
@@ -517,6 +504,19 @@ class VariationalSmootherConstraint : public System::Constraint
   virtual ~VariationalSmootherConstraint() override;
 
   virtual void constrain() override;
+
+  /**
+   * Given a mesh and a node in the mesh, the vector will be filled with
+   * every node directly attached to the given one. IF NO neighbors are found,
+   * all nodes on the containing side are treated as neighbors. This is useful
+   * when the node does not lie on an edge, such as the central face node in
+   * HEX27 elements.
+   */
+  static void find_nodal_or_face_neighbors(
```

Comment:
I hate moving things to `public:` before we need to, lest we find that someone starts using them and we need to change the API and now we have to worry about a transition path for compatibility.  It doesn't look to me like we'll need to change this API, but what do you think?